File: testdata/jennies/builders/map_of_builders/PHPBuilder/src/MapOfBuilders/DashboardBuilder.php
<?php

namespace Grafana\Foundation\MapOfBuilders;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\MapOfBuilders\Dashboard>
 */
class DashboardBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\MapOfBuilders\Dashboard $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\MapOfBuilders\Dashboard();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\MapOfBuilders\Dashboard
     */
    public function build()
    {
        return $this->internal;
    }

    /**
     * @param array<string, \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\MapOfBuilders\Panel>> $panels
     */
    public function panels(array $panels): static
    {
            $panelsResources = [];
            foreach ($panels as $key1 => $val1) {
                    $panelsResources[$key1] = $val1->build();
            }
        $this->internal->panels = $panelsResources;
    
        return $this;
    }

}

File: testdata/jennies/builders/map_of_builders/CSharpBuilder/src/Grafana/Foundation/MapOfBuilders/DashboardBuilder.cs
namespace Grafana.Foundation.MapOfBuilders;


public class DashboardBuilder : Cog.IBuilder<Dashboard>
{
    protected readonly Dashboard @internal;

    public DashboardBuilder()
    {
        this.@internal = new Dashboard();
    }

    public DashboardBuilder Panels(Dictionary<string, Cog.IBuilder<Panel>> panels)
    {
        Dictionary<string, Panel> panelsResources = new Dictionary<string, Panel>();
        foreach (var entry1 in panels)
        {
                Panel panelsDepth1 = entry1.Value.Build();
                panelsResources[entry1.Key] = panelsDepth1;
        }
        this.@internal.Panels = panelsResources;
        return this;
    }

    public Dashboard Build()
    {
        return this.@internal;
    }
}
